### ISSUE TITLE
Suggested enhancement: allow VimwikiGenerateTags to include context.

### ISSUE THREAD

**Issue by jean** [vimwiki/vimwiki#797]
Currently, `:VimwikiGenerateTags` can tag a whole file, a header in a file, or a standalone tag. That's pretty cool.
It would be even nicer if the generated tag index could optionally include some context, e.g. 

> == SomeTag ==
    - The tagged header: [[SomePage#SomeTag]]
    - "... let's tag this passage as [[OtherPage#SomeTag]] bla bla ..."
    - "... this is another aspect of [[ThisPage#SomeTag]] bla bla ..."

Another possibility is to include preceding and/or following lines of context.

**Comment by nathanbraun** [comment]
I agree, this would be awesome, for an example (all the links at the bottom have context) see:

https://notes.andymatuschak.org/About_these_notes

He even has a note that's roughly about this:

https://notes.andymatuschak.org/z7pGUpz2fQsHHUPbjThz85xXPvHwrmikAeYH4
